src/shared/components/plots/WaterfallPlot.tsx

```diff
@@ -0,0 +1,545 @@
+import _ from "lodash";
+import * as React from "react";
+import {observer, Observer} from "mobx-react";
+import bind from "bind-decorator";
+import {computed, observable} from "mobx";
+import CBIOPORTAL_VICTORY_THEME from "../../theme/cBioPoralTheme";
+import Timer = NodeJS.Timer;
+import {VictoryChart, VictoryAxis, VictoryBar, VictoryScatter, VictoryLegend, VictoryLabel} from "victory";
+import { makeScatterPlotSizeFunction as makePlotSizeFunction, dataPointIsLimited } from "./PlotUtils";
+import WaterfallPlotTooltip from "./WaterfallPlotTooltip";
+import { tickFormatNumeral } from "./TickUtils";
+import { IWaterfallPlotData, IAxisLogScaleParams } from "pages/resultsView/plots/PlotsTabUtils";
+import LetterIcon from "../cohort/LetterIcon";
+
+// TODO make distinction between public and internal interface for waterfall plot data
+export interface IBaseWaterfallPlotData {
+    value:number;
+    thresholdType?:'>'|'<'|undefined;
+    order?:number|undefined;
+    offset?:number;
+    fill?:string;
+    fillOpacity?:number;
+    stroke?:string;
+    strokeOpacity?:number;
+    strokeWidth?:number;
+    symbol?:string;
+    labelx?:number;
+    labely?:number;
+    labelVisibility?:boolean;
+    searchindicatorx?:number;
+    searchindicatory?:number;
+}
+
+export interface IWaterfallPlotProps<D extends IBaseWaterfallPlotData> {
+    svgId?:string;
+    title?:string;
+    data: D[];
+    chartWidth:number;
+    chartHeight:number;
+    highlight?:(d:D)=>boolean;
+    size?:number | ((d:D, active:boolean, isHighlighted?:boolean)=>number);
+    fill?:string|((d:D)=>string);
+    stroke?:string|((d:D)=>string);
+    fillOpacity?:number|((d:D)=>number);
+    strokeOpacity?:number|((d:D)=>number);
+    strokeWidth?:number|((d:D)=>number);
+    symbol?:string|((d:D)=>string);
+    labelVisibility?:boolean|((d:D)=>boolean);
+    zIndexSortBy?:((d:D)=>any)[]; // second argument to _.sortBy
+    tooltip?:(d:D)=>JSX.Element;
+    horizontal:boolean;
+    legendData?:{name:string|string[], symbol:any}[]; // see http://formidable.com/open-source/victory/docs/victory-legend/#data
+    log?:IAxisLogScaleParams|undefined;
+    useLogSpaceTicks?:boolean; // if log scale for an axis, then this prop determines whether the ticks are shown in post-log coordinate, or original data coordinate space
+    axisLabel?:string;
+    fontFamily?:string;
+    sortOrder:string|undefined;
+    pivotThreshold?:number;
+}
+
+const DEFAULT_FONT_FAMILY = "Verdana,Arial,sans-serif";
+export const LEGEND_Y = 30
+const RIGHT_PADDING = 120; // room for correlation info and legend
+const NUM_AXIS_TICKS = 8;
+const PLOT_DATA_PADDING_PIXELS = 50;
+const MIN_LOG_ARGUMENT = 0.01;
+const LEFT_PADDING = 25;
+const LABEL_OFFSET_FRACTION = .02;
+const LABEL_SIZE_MULTIPLIER = 1.5;
+const labelStyle = {
```

Comment:
why is this called labelStyle it doesn't seem to describe labels

Reply:
I agree. In the current state `label` means `searchIndicator`. Do you agree with renaming all occurences of `label` into `searchIndicator`?